File: clientapp/src/app/models/problem.ts
export interface Problem {
  id: number;
  sourceLanguage: string;
  targetLanguage: string;
  sourceCode: string;
  expectedTargetCode: string;
  explanation?: string;
  hint?: string;
}

export interface Submission {
  userCode: string;
  sourceLanguage: string;
  targetLanguage: string;
  originalSourceCode: string;
}

export interface CheckAnswerResponse {
  message: string;
  isCorrect: boolean;
  note?: string;
}

File: backend/Models/Problem.cs
namespace backend.Models
{
    public class Problem
    {
        public int Id { get; set; }
        public string SourceLanguage { get; set; }
        public string TargetLanguage { get; set; }
        public string SourceCode { get; set; }
        public string ExpectedTargetCode { get; set; }
        public string Explanation { get; set; }
        public string Hint { get; set; }
    }
    public class Submission
    {
        public string UserCode { get; set; }
        public string SourceLanguage { get; set; }
        public string TargetLanguage { get; set; }
        public string OriginalSourceCode { get; set; }
    }
}
